pkg/vault/vault_paths.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package vault
+
+import "fmt"
+
+// The path of the Vault plugin to use to retrieve dataset credentials stored in kubernetes secret.
+// vault-plugin-secrets-kubernetes-reader plugin is used for this purpose and is enabled
+// in kubernetes-secrets path. (https://github.com/mesh-for-data/vault-plugin-secrets-kubernetes-reader)
+// TODO: pass the plugin path in m4d-config ConfigMap
+const vaultPluginPath = "kubernetes-secrets"
+
+// PathForReadingKubeSecret returns the path to Vault secret that holds dataset credentials
+// stored in kubernetes secret.
+// Vault plugin vault-plugin-secrets-kubernetes-reader is used for reading kubernetes secret
+// (https://github.com/mesh-for-data/vault-plugin-secrets-kubernetes-reader)
+// The path contains the following parts:
+// - pluginPath is the Vault path where vault-plugin-secrets-kubernetes-reader plugin is enabled.
+// - secret name
+// - secret namespace
+// for example, for secret name my-secret and namespace default it will be of the form:
+// "/v1/kubernetes-secrets/my-secret?namespace=default"
+func PathForReadingKubeSecret(secretNamespace string, secretName string) string {
```

Comment:
Is this path the one that the modules are using to look up the secret in vault? Does the namespace as query parameter work out of the box or do we have to change something in the modules?

Reply:
Yes, this is the path to the secret. The kubernetes-secrets plugin reads a k8s secret identified by the secret name and namespace.